### PR TITLE
Enable UT test in nightly with On-demand test and On-demand test for public pytorch without torch-xpu-ops update

### PR DESCRIPTION
**Pull request by RUIJIEZHONG66166**

### CODE REVIEW

File: .github/actions/inductor-xpu-ut-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        for ut_suite in $(echo ${{ inputs.ut_suite }} |sed 's/,/ /g')
+        do
+          contains "op_example,op_finegrain,op_ut,torch_xpu" $ut_suite
+          $contains_status
+          if [[ ${ut_suite} == 'op_example' ]]; then
+            unset PYTORCH_TEST_WITH_SLOW
+            unset PYTORCH_ENABLE_XPU_FALLBACK
+            cd ${{ github.workspace }}
+            cd examples
+            timeout 8000 pytest -v 2>&1 | tee ${{ github.workspace }}/ut_log/XPU_OP_Examples_UT_Test.log
```
Comment (chuanqi129) [inline .github/actions/inductor-xpu-ut-test/action.yml:41]: This method will block test failure return as job status. E.g. if the test failed, the job status will be success anyway. Another method is add post-op log analysis

File: .github/workflows/inductor_xpu_e2e_nightly.yml [hunk 2]
```diff
@@ -200,6 +218,71 @@ jobs:
...
+              rm -rf third_party/torch-xpu-ops && cp -r ../torch-xpu-ops third_party/
+            else
+              echo "Not update torch-xpu-ops"
+            fi
+          fi
+          # Workaround for torch-xpu-ops ci test
+          sed -i "s/checkout --quiet \${TORCH_XPU_OPS_COMMIT}/log -n 1/g" caffe2/CMakeLists.txt
+      - name: Build Pytorch XPU
+        run: |
+          source activate xpu_op_${ZE_AFFINITY_MASK}
```
[the remark this answers is not in the document]
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_e2e_nightly.yml:251]: Done

File: .github/workflows/inductor_xpu_e2e_nightly.yml [hunk 3]
```diff
@@ -200,6 +218,71 @@ jobs:
           scenario: ${{ inputs.scenario }}
           hf_token: ${{ secrets.HUGGING_FACE_HUB_TOKEN }}
 
+  Inductor-XPU-UT-Nightly-Tests:
+    runs-on: linux.idc.xpu
+    timeout-minutes: 900
```
[the remark this answers is not in the document]
Comment (RUIJIEZHONG66166) [reply .github/workflows/inductor_xpu_e2e_nightly.yml:235]: Done

File: .github/actions/inductor-xpu-ut-test/action.yml (hunk 1 above)
Comment (RUIJIEZHONG66166) [reply .github/actions/inductor-xpu-ut-test/action.yml:41]: Sure. Does it check with fail case number? It mean if fail case num > expected num, return false; else return success. 
If so, could you provide the expected num for these UT cases. Thanks~

**Comment by fengyuan14** [comment]
Stop the preci since there is a conflict to resolve.

**Comment by chuanqi129** [comment]
close it because the function has enabled by other PRs

**Pull request closed without merging**